addons/mail/static/tests/mail_utils.test.js

```diff
@@ -205,6 +206,25 @@ test("url with number in subdomain", async () => {
     );
 });
 
+test("parseEmail utility function", async () => {
+    const attackPayloads = [
+        '"'.repeat(20) + "<".repeat(100000) + "@example.com>",
+        " <" + "@".repeat(100000) + "@",
+        "<@".repeat(100000) + "@",
+    ];
+
+    for (const [i, attack] of attackPayloads.entries()) {
+        const start = performance.now();
+        const result = parseEmail(attack);
+        const end = performance.now();
+        const duration = end - start;
+
+        console.log(`Test ${i + 1}: duration = ${duration.toFixed(2)} ms`);
+        expect(duration < 1000).toBe(true);
```

Comment:
This is prone to failing on our CI infrastructure because we don't have guarantee of CPU time, and we don't want this to fail randomly.

What is the expected usual time under good conditions, and the time the same test took before the fix?

I'd consider at least a value of ~5s to be on the safer side, if the failing time is much higher than data.